wgpu-core/src/lock/mod.rs

```diff
@@ -31,11 +31,17 @@ pub mod rank;
 #[cfg_attr(not(wgpu_validate_locks), allow(dead_code))]
 mod ranked;
 
-#[cfg_attr(wgpu_validate_locks, allow(dead_code))]
+#[cfg(feature = "observe_locks")]
+mod observing;
+
+#[cfg_attr(any(wgpu_validate_locks, feature = "observe_locks"), allow(dead_code))]
 mod vanilla;
 
 #[cfg(wgpu_validate_locks)]
 pub use ranked::{Mutex, MutexGuard};
 
-#[cfg(not(wgpu_validate_locks))]
+#[cfg(feature = "observe_locks")]
+pub use observing::{Mutex, MutexGuard};
```

Comment:
I just want conditional compilation and convenience. Do you think a `cfg` would be more suitable? 

It's hard to see `wgpu-core` users somehow explicitly requesting `observe_locks` in their dependencies, so the fact that it's exposed doesn't seem very important.

Reply:
Yeah, I think `cfg` would be better here, and it would be consistent with the existing `cfg(wgpu_validate_locks)`. I'd prefer users never even see the `feature` in their `Cargo.toml` completion in IDEs.

I won't hard-block on this, tho.